File: Flora Dummy/Page_ReadVC.m
//
//  Page_ReadVC.m
//  Flora Dummy
//
//  Created by Zach Nichols on 11/9/13.
//  Copyright (c) 2013 SGSC. All rights reserved.
//

#import "Page_ReadVC.h"

@interface Page_ReadVC ()
{
    
}

@end

@implementation Page_ReadVC
@synthesize summaryTextView, pageText;

- (id)initWithNibName:(NSString *)nibNameOrNil bundle:(NSBundle *)nibBundleOrNil
{
    self = [super initWithNibName:nibNameOrNil bundle:nibBundleOrNil];
    if (self) {
        // Custom initialization
    }
    return self;
}

- (void)viewDidLoad
{
    [super viewDidLoad];
    // Do any additional setup after loading the view from its nib.
    
    //summaryTextView = [[UITextView alloc] initWithFrame:CGRectMake(33, 72, 971, 602)];
    summaryTextView.text = pageText;
    
    super.titleLabel.frame = CGRectMake(super.titleLabel.frame.origin.x,
                                 super.titleLabel.frame.origin.y,
                                 super.titleLabel.frame.size.width,
                                 super.dateLabel.frame.size.height); // on purpose
    super.titleLabel.font = super.dateLabel.font;
    
    super.dateLabel.frame = CGRectMake(super.dateLabel.frame.origin.x,
                                 20,
                                 super.dateLabel.frame.size.width,
                                 super.dateLabel.frame.size.height);
    super.otherLabel.frame = CGRectMake(super.otherLabel.frame.origin.x,
                                 20,
                                 super.otherLabel.frame.size.width,
                                 super.otherLabel.frame.size.height);

}

- (void)didReceiveMemoryWarning
{
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}


-(void)updateColors
{
    [super updateColors];
    
    [self outlineTextInTextView:summaryTextView];
    summaryTextView.textColor = primaryColor;
    summaryTextView.backgroundColor = secondaryColor;
    
}

@end

File: Flora Dummy/Page_ReadVC.swift
//
//  Page_ReadVC.swift
//  FloraDummy
//
//  Created by Michael Schloss on 10/31/14.
//  Copyright (c) 2014 SGSC. All rights reserved.
//

import UIKit

class Page_ReadVC: FormattedVC
{
    var pageText : NSString?
    
    @IBOutlet var summaryTextView : UITextView?

    override func viewDidLoad()
    {
        super.viewDidLoad()

        let formattedString = (pageText! as NSString).stringByReplacingOccurrencesOfString("\\n", withString: "\n") as String
        
        summaryTextView!.text = formattedString
    }

    //Color Theme setting
    override func viewWillAppear(animated: Bool)
    {
        super.viewWillAppear(animated)
        
        Definitions.outlineTextInTextView(summaryTextView!, forFont: summaryTextView!.font)
        
        summaryTextView!.textColor = primaryColor
        summaryTextView!.backgroundColor = secondaryColor
    }
}
